plugin/state.go

```diff
@@ -100,20 +100,29 @@ func awsCatalogPersistedStateFromProto(in *pb.HostCatalogPersisted, opts ...awsC
 		return nil, errors.New("missing persisted secrets")
 	}
 
+	unknownFields := structFields(data)
+
 	accessKeyId, err := getStringValue(data, constAccessKeyId, true)
 	if err != nil {
 		return nil, fmt.Errorf("persisted state integrity error: %w", err)
 	}
+	delete(unknownFields, constAccessKeyId)
 
 	secretAccessKey, err := getStringValue(data, constSecretAccessKey, true)
 	if err != nil {
 		return nil, fmt.Errorf("persisted state integrity error: %w", err)
 	}
+	delete(unknownFields, constSecretAccessKey)
 
 	credsLastRotatedTime, err := getTimeValue(data, constCredsLastRotatedTime)
 	if err != nil {
 		return nil, fmt.Errorf("persisted state integrity error: %w", err)
 	}
+	delete(unknownFields, constCredsLastRotatedTime)
+
+	if len(unknownFields) != 0 {
```

Comment:
If the persisted data contains data we don't know about and everything else works fine, I think we should probably just ignore the extra data and clear the unknown data the next time a method that allows us to update the persisted data is called.

Reply:
:+1: sounds good here, will make the change. For sure agree especially on the basis that the user really doesn't control this. Think my rationale here was to enforce integrity as much as possible.

Reply:
Done!